image/image.go

```diff
@@ -213,19 +213,24 @@ func makeFetcher(sto Store, dlOpts *DownloadOptions, db *asserts.Database) *addi
 func installCloudConfig(gadgetDir string) error {
 	var err error
 
-	cloudDir := filepath.Join(dirs.GlobalRootDir, "/etc/cloud")
+	cloudDir := filepath.Join(dirs.GlobalRootDir, "/etc/cloud/")
 	if err := os.MkdirAll(cloudDir, 0755); err != nil {
 		return err
 	}
 
 	cloudConfig := filepath.Join(gadgetDir, "cloud.conf")
 	if osutil.FileExists(cloudConfig) {
-		dst := filepath.Join(cloudDir, "cloud.cfg")
+		cfgDir := filepath.Join(dirs.GlobalRootDir, "/etc/cloud/cloud.cfg.d")
+		if err := os.MkdirAll(cfgDir, 0755); err != nil {
+			return err
+		}
+		dst := filepath.Join(cfgDir, "10_gadget_snap_cloud.cfg")
```

Comment:
@niemeyer Thanks for this input. I asked Ryan to answer this, this branch implements his recommendations as the cloud-init upstream. My initial work was simply to take over /etc/cloud/ but the cloud-init upstreams recommended against it:

"""
This isn't sufficient unless we want to put a high burden on the gadget snaps to understand how cloud-init behaves by default. Rather, when you install cloud-init, /etc/cloud is populated with a default configuration; this should always remain in the os-snap; it can remain in addition to a cloud-init.disabled to be off by default, but the gadget snaps should only include any changes they want to make from the default rather than having to carry all of the default cloud-init behavior driven from the packaged config
"""